orttraining/orttraining/python/training/_ortmodule_training_manager.py

```diff
@@ -32,13 +32,13 @@ def forward(self, *inputs, **kwargs):
         # Exporting module to ONNX for the first time
         build_gradient_graph = self._export_model(*inputs, **kwargs)
         if build_gradient_graph:
-            # If model was exported, then initialize the graph builder
-            self._initialize_graph_builder(training=True)
-
             # Save the onnx model if the model was exported
             if self._save_onnx:
                 onnx.save(self._onnx_model, self._save_onnx_prefix + '_exported_training_model.onnx')
 
+            # If model was exported, then initialize the graph builder
+            self._initialize_graph_builder(training=True)
```

Comment:
Does this change do anything? I would have expected that we need to initialize the graph builder before saving a graph called "training_model" (but might be getting confused about the names...).

Reply:
I just switched places with `onnx.save()`. It is quiet common that the inference model is exported but the gradient building fails. So I switched places to at least get the inference graph saved

Reply:
In that case, can you save both, and use a different filename for the graph saved before building the training graph?

Reply:
We saved several versions already
XXX_exported.onnx as exported by pytorch onnx exporter
XXX_exported_training_model.onnx as the training graph
XXX_inference_optimized.onnx as the optimized inference model (from graph builder)
XXX_training_optimized.onnx  as the optimized training model (from graph builder)

Reply:
Ah, I think I understand now. The same graph will be saved in either version (because `self._onnx_model` is the same before and after the call to `self._initialize_graph_builder()`), but now it will happen regardless of whether `self._initialize_graph_builder()` succeeds or not.

My confusion arose from the name "training_model", which doesn't sound like an inference graph :).

Makes sense now, thanks!